pulsar-client-cpp/lib/MultiTopicsConsumerImpl.cc

```diff
@@ -572,6 +572,46 @@ void MultiTopicsConsumerImpl::acknowledgeAsync(const MessageId& msgId, ResultCal
     }
 }
 
+void MultiTopicsConsumerImpl::acknowledgeAsync(const MessageIdList& messageIdList, ResultCallback callback) {
+    if (state_ != Ready) {
+        callback(ResultAlreadyClosed);
+        return;
+    }
+
+    std::map<std::string, MessageIdList> topicToMessageId;
+    for (const MessageId& messageId : messageIdList) {
+        auto topicName = messageId.getTopicName();
+        if (topicToMessageId.count(topicName) == 0) {
+            topicToMessageId.emplace(topicName, std::vector<MessageId>());
+        }
+        topicToMessageId[topicName].emplace_back(messageId);
+    }
+
+    std::shared_ptr<std::atomic<int>> needCallBack =
+        std::make_shared<std::atomic<int>>(topicToMessageId.size());
```

Comment:
```suggestion
    auto needCallBack = std::make_shared<std::atomic<int>>(topicToMessageId.size());
```

For complicated types, it's better to use C++'s auto type deduction.